homeassistant/components/point/binary_sensor.py

```diff
@@ -124,3 +120,18 @@ def is_on(self):
             # connectivity is the other way around.
             return not self._is_on
         return self._is_on
+
+    @property
+    def name(self):
+        """Return the display name of this device."""
+        return f"{self._name} {self._device_name.capitalize()}"
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return DEVICE_TYPES.get(self._device_name, {}).get("icon", super().icon)
```

Comment:
Which super class' `icon` property do we fall back to here, it doesn't seem to be implemented neither in `BinarySensorEntity` nor in `MinutPointEntity`, and we `Entity` just returns `None`. If not needed, just do:
```suggestion
        return DEVICE_TYPES.get(self._device_name, {}).get("icon")
```

Reply:
I was hoping that `super().icon` refers to the device_class icon.

Reply:
No, but if `icon` is `None` the device class' icon will be used.